### DISCUSSION TITLE
Pyright doesn't recognize that PySide6.QtCore.QDate can be ordered

### DISCUSSION THREAD

**Discussion by ayhon** [microsoft/pyright#5645] [Q&A]
I've filed a [bug report with the `PySide6` project](https://bugreports.qt.io/browse/PYSIDE-2408), but I'm unsure if this is actionable on their side or perhaps more of a bug with `pyright`.

Given the following file
```python
from PySide6.QtCore import QDate
QDate(2023, 7, 30) < QDate(2023, 8, 1)
```
if I run `pyright` I get the following
```shell
No configuration file found.
No pyproject.toml file found.
stubPath /home/ayhon/Decay/2023-08-04/typings is not a valid directory.
Assuming Python platform Linux
Searching for source files
Found 1 source file
/home/ayhon/Decay/2023-08-04/qdate_orderable.py
  /home/ayhon/Decay/2023-08-04/qdate_orderable.py:2:1 - error: Operator "<" not supported for types "QDate" and "QDate"
    Operator "<" not supported for types "QDate" and "QDate" (reportGeneralTypeIssues)
1 error, 0 warnings, 0 infos 
Completed in 0.897sec
```
However, `PySide6.QtCore.QDate` *is* orderable. 
```python
>>> from PySide6.QtCore import QDate
>>> d1 = QDate(2023, 7, 30)
>>> d1
PySide6.QtCore.QDate(2023, 7, 30)
>>> d2 = QDate(2023, 8, 1)
>>> d2
PySide6.QtCore.QDate(2023, 8, 1)
>>> d1 < d2
True
```

How does `pyright` detect if a class is orderable?

**Comment by erictraut** [comment]
This is not a pyright bug. The class `QDate` does not define a `__lt__` or `__gt__` magic method, so static type checkers necessarily assume that the class does not support these methods. This looks like a straightforward bug in the type stub.
